src/url.js

```diff
@@ -89,61 +78,75 @@ export function getWinOrigin(win) {
  * the URL gets resolved.
  * Consider the returned object immutable. This is enforced during
  * testing by freezing the object.
+ * TODO(#34453): The URL constructor isn't supported in IE11, but is supported
+ * everywhere else. There's a lot of code paths (and all uses of the LruCache)
+ * that are built around this polyfill. Once we can drop IE11 support and just
+ * use the URL constructor, we can clear out all of parseWithA, all the URL
+ * cache logic (incl. additional caches in other call-sites). Most is guarded by
+ * IS_ESM and is only included in nomodule builds, but still.
  * @param {string} url
  * @param {boolean=} opt_nocache
  *   Cache is always ignored on ESM builds, see https://go.amp.dev/pr/31594
  * @return {!Location}
  */
 export function parseUrlDeprecated(url, opt_nocache) {
-  if (!a) {
-    a = /** @type {!HTMLAnchorElement} */ (self.document.createElement('a'));
-    cache = IS_ESM
+  if (!cachedAnchorEl) {
+    cachedAnchorEl = /** @type {!HTMLAnchorElement} */ (
+      self.document.createElement('a')
+    );
+    urlCache = IS_ESM
       ? null
       : self.__AMP_URL_CACHE || (self.__AMP_URL_CACHE = new LruCache(100));
   }
 
-  return parseUrlWithA(a, url, IS_ESM || opt_nocache ? null : cache);
+  return parseUrlWithA(
+    cachedAnchorEl,
+    url,
+    IS_ESM || opt_nocache ? null : urlCache
```

Comment:
Just initialize `let urlCache = null`, and we can remove this ternary.
```suggestion
    urlCache
```

Reply:
I'm not certain that has the desired effect, so I'm going to punt on that until my next PR moving the URL helpers. opt_nocache is used by some callers and seems to matter.